created a google login

iraskiba opened the pull request:
created a store for storing data about the client and the token, 
adding a request to google api to get data about an authorized user, 
created a hook to verify an authorized user,
fixed modal windows errors

iraskiba committed:
created a google login

created a store for storing data about the client and the token,
adding a request to google api to get data about an authorized user,
created a hook to verify an authorized user,
fixed modal windows errors

iraskiba committed:
added persist store

added persist store

iraskiba committed:
added modal window component

added modal window component that displays dynamic content according to the locale.

Khomko-Dima commented on src/app/App.tsx:
```diff
@@ -13,8 +14,10 @@ function App() {
     <>
-      <NotificationDisplay />
-      {routing}
+      <GoogleOAuthProvider clientId="567602328922-q58mvvbists8jqub7ue0ove59e9nut4c.apps.googleusercontent.com">
```
clientId бери из env переменной

Khomko-Dima commented on src/pages/login/model/auth-hook.ts:
```diff
@@ -0,0 +1,9 @@
+
+const useIsAuthenticated = () => {
+  const { user } = useUserStore()
+  const accessToken = user?.accessToken || sessionStorage.getItem('accessToken')
```
Бери всегда из store, т.к он будет возвращать по умолчнию синхронизированную с sessionStorage

Khomko-Dima commented on src/shared/ui/custom-button-plus/custom-button-plus.tsx:
```diff
@@ -6,11 +6,12 @@ import { ButtonProps } from 'antd/es/button/button'
 
 const buttonTexts = {
   new: 'Add New',
+  task: 'Add New Task',
```
А что если появятси ещё 5 новых версий кнопок?
Надо менять логику)

Khomko-Dima commented on src/widgets/header/ui/modal-content.tsx:
```diff
@@ -0,0 +1,103 @@
+  onSubmit: (data: Record<string, any>) => void
+}
+
+export const ModalContentMap: FC<ModalContentProps> = ({ path, onSubmit }) => {
```
Тут такое точно созвать нельзя
По сути мыслишь не в правильном направлении.
Ты делаешь так что шапка сайта отвечает за всю логику системы
Скил матрица страдает из-за этого
По сути норм вариант в кажой из из страниц создать форму редактирвоания или для создания
А ещё лучше сделать в папке enteties сущности
например
enteties/deal
enteties/customer
Внутри
такая структура
enteties
-deal
--ui
---form
---list
--api
--model
--types
-index.ts (тут экспорт тех сущностей каторые используются вне папки enteties)
И дале сделать модалку через eventBus. и в момент emit модалки передавать компонент каторый модалка должна открыть
Вот пример моей модалки, она может открыть 2 штуки (по сути у тебя будет проще)
`
type ModalType = { component: React.ReactNode; } & ModalProps;
const ModalComponent = () => {
	const [isOpen, setIsOpen] = useState({ first: false, second: false });
	const [data, setData] = useState<StateData>(initialState);
	const OpenModal = (data: ModalType) => {
		setIsOpen((prev) => {
			if (prev.first) return { ...prev, second: true };
			return { first: true, second: false };
		});
		setData((prev) => {
			if (prev.first) return { ...prev, second: data };
			return { second: null, first: data };
		});
	};
	const CloseModal = (all?: boolean) => {
		if (all) {
			setIsOpen({ first: false, second: false });
			setData(initialState);
		} else {
			setIsOpen((prev) => {
				if (prev.second) return { ...prev, second: false };
				return { first: false, second: false };
			});
			setData((prev) => {
				if (prev.second) return { ...prev, second: null };
				return { second: null, first: null };
			});
		}
	};
	useEffect(() => {
		const unsubscribe = ModalEvent.subscribeOpen(OpenModal);
		return () => unsubscribe();
	}, []);

	useEffect(() => {
		const unsubscribe = ModalEvent.subscribeClose(CloseModal);
		return () => unsubscribe();
	}, []);

	return (
		<>
			<ModalContent isOpen={isOpen.first} data={data.first} closeModal={CloseModal} />
			<ModalContent isOpen={isOpen.second} data={data.second} closeModal={CloseModal} />
		</>
	);
 };
`

iraskiba committed:
added modal opening manager

made component-manager of opening tasks on basis of tasks component
moved each modal-component to the corresponding folder in entities

iraskiba committed:
added opening modal by eventBus

modified modal window accepting a child,
change buttons open modal windows via event buss,
changed the paths for the events page,
added day picker for modal deals window,
each modal now accepts the record and displays the content for editing

iraskiba committed:
added store for storing customers- data

added store for storing customers- data and displaying it on the dashboard content page

iraskiba committed:
added modal window add new

added modal window "add new" for dashboard

iraskiba committed:
added modal Edit Activity and Select Customer

added modal Edit Activity and Select Customer

iraskiba committed:
added sort by date, changed types and created type files

added sort by date, changed types and created type-files, added stores

iraskiba committed:
added new page - deal details

iraskiba committed:
added states for view-all buttons, styles moved to style files

Khomko-Dima commented on src/app/router/index.tsx:
```diff
@@ -6,11 +6,28 @@ import { Customers } from '@pages/customers'
+import { Login } from '@pages/login'
+import { useIsAuthenticated } from '@pages/login'
+import { ReactNode } from 'react'
+import DealDetails from '@pages/deal-details/ui/deal-details.tsx'
```
импорт сократи

Khomko-Dima commented on src/enteties/activity/ui/modal-edit-activity.tsx:
```diff
@@ -0,0 +1,61 @@
+import { FormProvider, useForm } from 'react-hook-form'
+import { eventBus } from '@shared/lib/event-bus.ts'
+import { ModalEvent } from '@process/modal'
+import DateInput from '../../deals/ui/day-picker.tsx'
```
и тут

Khomko-Dima commented on src/enteties/activity/ui/modal-edit-activity.tsx:
```diff
@@ -0,0 +1,61 @@
+      <form onSubmit={onSubmit}>
+        <Input.TextArea
+          rows={4}
+          placeholder="Evaluation and removal of the old system"
```
данные из этого поля попадают в форму?

iraskiba replied:
уже да)

Khomko-Dima commented on src/enteties/activity/ui/modal-edit-activity.tsx:
```diff
@@ -0,0 +1,61 @@
+        />
+        <DateInput />
+
+        <div
```
замени div на элемент Flex из antd

Khomko-Dima commented on src/enteties/activity/ui/modal-edit-activity.tsx:
```diff
@@ -0,0 +1,61 @@
+        <Button
+          onClick={handleClose}
+          type="text"
+          style={{ marginRight: '8px', color: 'red' }}
```
лучше стилизовать в файле стилей

Khomko-Dima commented on src/enteties/customers/ui/modal-add-customers.tsx:
```diff
@@ -0,0 +1,119 @@
+import { FormEvent, useState } from 'react'
+import { eventBus } from '@shared/lib/event-bus.ts'
+import { FormProvider, useForm } from 'react-hook-form'
+import FormInput from '@shared/ui/form-items/input'
```
сократи с import FormInput from '@shared/ui/form-items/input'
до 
import {FormInput } from '@shared/ui/form-items'

Khomko-Dima commented on src/enteties/customers/ui/modal-add-customers.tsx:
```diff
@@ -0,0 +1,119 @@
+import { UploadOutlined } from '@ant-design/icons'
+import styles from './styles.module.scss'
+
+import { ModalEvent } from '../../../process/modal/index.ts'
```
импорт!

Khomko-Dima commented on src/enteties/customers/ui/modal-add-customers.tsx:
```diff
@@ -0,0 +1,119 @@
+    handleClose()
+  }
+
+  const myPromiseUpload = (file: File): Promise<string> => {
```
вынеси из компонеты чтоб не пересозавать каждый раз функцию

Khomko-Dima replied:
Можно вынести в shared/lib

Khomko-Dima commented on src/enteties/customers/ui/select-customer.tsx:
```diff
@@ -0,0 +1,55 @@
+import { Avatar, Button } from 'antd'
+import styles from './styles.module.scss'
+import useCustomerStore from '@pages/customers/model/customers-store.ts'
```
сократи, т.к мы внутри этой же структуры допускается вид
import useCustomerStore from '../../model/customers-store.ts'
и тогда лучше
import {useCustomerStore } from '../../model'

Khomko-Dima commented on src/enteties/dashboard/ui/modal-add-new.tsx:
```diff
@@ -0,0 +1,45 @@
+  UsergroupDeleteOutlined,
+  ArrowRightOutlined,
+} from '@ant-design/icons'
+import { ModalEvent } from '../../../process/modal/index.ts'
```
импорт

Khomko-Dima commented on src/enteties/deals/ui/day-picker.tsx:
```diff
@@ -0,0 +1,33 @@
+import { useState } from 'react'
+import { DayPicker } from 'react-day-picker'
+import 'react-day-picker/dist/style.css'
+import FormInput from '@shared/ui/form-items/input'
```
импорт

Khomko-Dima commented on src/enteties/deals/ui/modal-add-deals.tsx:
```diff
@@ -0,0 +1,137 @@
+import DateInput from './day-picker.tsx'
+import UniversalSelect from '@shared/ui/form-items/select'
+import { OptionProps } from 'rc-select/lib/Option'
+import { ModalEvent } from '../../../process/modal/index.ts'
```
импорт

Khomko-Dima commented on src/enteties/deals/ui/modal-add-deals.tsx:
```diff
@@ -0,0 +1,137 @@
+import { FormEvent } from 'react'
+import { eventBus } from '@shared/lib/event-bus.ts'
+import { FormProvider, useForm } from 'react-hook-form'
+import FormInput from '@shared/ui/form-items/input'
```
для всех элементов фолрмы сократи до
import {} from '@shared/ui/form-items'

Khomko-Dima commented on src/enteties/tasks/ui/modal-add-tasks.tsx:
```diff
@@ -0,0 +1,48 @@
+import FormInput from '@shared/ui/form-items/input'
+import { FormProvider, useForm } from 'react-hook-form'
+import { eventBus } from '@shared/lib/event-bus.ts'
+import { ModalEvent } from '../../../process/modal/index.ts'
```
импорт

Khomko-Dima commented on src/enteties/tasks/ui/modal-add-tasks.tsx:
```diff
@@ -0,0 +1,48 @@
+import { Button, Input } from 'antd'
+import { FormEvent } from 'react'
+import FormInput from '@shared/ui/form-items/input'
```
импорт

iraskiba committed:
fix recomendation,install libs for fsd and changed import path

Khomko-Dima commented on src/entities/customers/ui/modal-add-customers.tsx:
```diff
@@ -0,0 +1,109 @@
+import { FormProvider, useForm } from 'react-hook-form'
+import { eventBus } from '@shared/lib/event-bus.ts'
+import { myPromiseUpload } from '@shared/lib/uploadFunction.ts'
+import { FormInput } from '@shared/ui/form-items/input'
```
import { FormInput } from '@shared/ui/form-items'

iraskiba committed:
added user login

and displaying the user's icon.

iraskiba committed:
fixed displaying the user's icon

iraskiba committed:
fixed displaying the user's icon in deals-details page
